### ISSUE TITLE
Case sensitivity with abook using notmuch address

### ISSUE THREAD

**Issue by mrichar1** [pazz/alot#1516]
I've set up alot to use `notmuch address` as the abook entry following the docs:

```
command = 'notmuch address --format=json --output=recipients date:1Y..'
regexp = '\[?{"name": "(?P<name>.*)", "address": "(?P<email>.+)", "name-addr": ".*"}[,\]]?'
shellcommand_external_filtering = False
ignorecase = True
```

When I 'search' by typing a partial name and hitting tab, it seems to be case-sensitive in what it searches for, despite setting `ignorecase`.

For example, I have several contacts called 'Bruce', with the following data in notmuch:

```
BROWN Bruce <bruce.brown@example.com>
Bruce Smith <Bruce.Duncan@example.com>
BRUCE David <David.Bruce@example.com>
```

When searching, `bruce` matches only the first one; `Bruce` matches all 3, and `BRUCE` matches only the last one. 

**Software Versions**
- Python version: 3.7.5
- Notmuch version:  0.29.1
- Alot version: github master

**Comment by pazz** [comment]
Thanks for reporting this.
In case you want to give it a try and submit a fix: the problem is that 
[the constructor of AdressBook Subclasses](https://github.com/pazz/alot/blob/master/alot/addressbook/external.py#L39) first set the value of `self.reflags` (by calling the super method, which interprets the `ignorecase` flat) and then, 3 lines below, overwrites this value with the default of the `reflags` parameter (0).

**Comment by mrichar1** [comment]
Fixed in #1517
